### PR TITLE
Fix archive sorting.

### PR DESCRIPTION
**Pull request by ajurasz**
This PR will fix sorting issue mentioned in #365.

### CODE REVIEW

File: src/main/java/com/jvm_bloggers/frontend/public_area/all_issues/AllIssuesPageBackingBean.java [hunk 1]
```diff
@@ -33,7 +34,11 @@ public AllIssuesPageBackingBean(NewsletterIssueForListingQuery query) {
                     .sorted(Comparator.comparing(NewsletterIssueForListing::getIssueNumber))
                     .reverse()
             );
-        return TreeMap.ofAll(issuesGroupedByYearMonth.toJavaMap());
```
Comment (ajurasz) [inline src/main/java/com/jvm_bloggers/frontend/public_area/all_issues/AllIssuesPageBackingBean.java]: Factory method by default use `EntryComparator.natural()` to sort keys - this is way months were is ASC order.

File: src/test/groovy/com/jvm_bloggers/frontend/public_area/all_issues/AllIssuesPageBackingBeanSpec.groovy [hunk 2]
```diff
@@ -39,7 +39,7 @@ class AllIssuesPageBackingBeanSpec extends Specification {
         then:
         groupedIssues.size() == 4
         javaslang.collection.Set<YearMonth> yearMonths = groupedIssues.keySet()
-        yearMonths.eq(javaslang.collection.HashSet.of(
+        yearMonths.eq(javaslang.collection.LinkedHashSet.of(
```
Comment (ajurasz) [inline src/test/groovy/com/jvm_bloggers/frontend/public_area/all_issues/AllIssuesPageBackingBeanSpec.groovy]: We need to keep the order of this map.

File: src/main/java/com/jvm_bloggers/frontend/public_area/all_issues/AllIssuesPageBackingBean.java [hunk 3]
```diff
@@ -33,7 +34,11 @@ public AllIssuesPageBackingBean(NewsletterIssueForListingQuery query) {
...
+        return TreeMap.ofEntries(
+            Comparator.reverseOrder(),
+            issuesGroupedByYearMonth
+                .map(Tuple2::new)
+                .toArray());
```
Comment (airborn) [inline src/main/java/com/jvm_bloggers/frontend/public_area/all_issues/AllIssuesPageBackingBean.java]: Can we move this mapping to the temp variable `issuesGroupedByYearMonth` just to have simple map creation over here?

**Comment by airborn** [comment]
:+1:

**Pull request merged**